public/scripts/main.js

```diff
@@ -273,38 +383,109 @@ container.addEventListener('keyup', function(event) {
   keys[event.keyCode] = false;
 });
 
+// Skills
+/*
+player.skillA1 = new Skills({
+  description: 'aa',
+  damage: -5,
+});
+*/
+player.skillA1 = skilldefault.Skill1;
+
+function fightLoop(subject, opponent) {
+  // assume player act first
+  if (subject.fightSpeed >= opponent.fightSpeed) {
+    if (inFightScene) {
+      doKeyfight(keys, opponent);
+      if (opponent.hp <= 0) {
+        inFightScene = false;
+      }
+    }
+    if (inFightScene) {
+      enemyfight(opponent, subject);
+      if (subject.hp <= 0) {
+        inFightScene = false;
+      }
+    }
+  } else {
+    if (inFightScene) {
+      enemyfight(opponent, subject);
+      if (subject.hp <= 0) {
+        inFightScene = false;
+      }
+    }
+    if (inFightScene) {
+      doKeyfight(keys, opponent);
+      if (opponent.hp <= 0) {
+        inFightScene = false;
+      }
+    }
+  }
+  fightSceneLoad(player, npc);
+}
+
+// Enemy fight strategy
+function enemyfight(opponent, subject) {
+  // alert(opponent.hp, subject.hp);
+  // opponent.skillA1.hpchange(subject, 0);
+}
+
+function doKeyfight(keys) {
+  // trigger escape with E
+  if (keys[32] || keys[69]) {
+    fightLayer.remove();
+    stage.add(layer);
+    inFightScene = false;
+  } else if (keys[81]) {
+  // press Q for normal hit
+  // normalattack.hpchange(npc, 0);
+    // player.skillA1.hpchange(npc, 0);
+    // npc.hp -= 50;
+
+    // console.log(player.skillA1);
+
+    player.skillA1.hpChange(npc, -50);
+
+    // alert(player.skillA1.descripttion);
+    // npc.skillA1.hpchange(-50);
+    console.log(npc.hp);
+  }
+}
+
 function doKeyProcess(keys) {
-  // Down arrow or W for moving sprite down
```

Comment:
change key case about in fight or not.
now it should only has movement and interact (maybe included Matt's inventory keys)